Assets/HoloToolkit/UX/Scripts/Buttons/ButtonIconProfileFont.cs

```diff
@@ -0,0 +1,399 @@
+
+                        iconFont = null;
+                    }
+                }
+                else
+                {
+                    // Check to see if our font has been outdated
+                    if (iconFont != null)
+                    {
+                        // If it's been outdated, destroy the font
+                        if (iconFont.name != OSFontName || iconFont.fontSize != FontSize)
+                        {
+                            // Destroy to prevent leaks
+                            if (Application.isPlaying)
+                                GameObject.Destroy(iconFont);
+                            else
+                                GameObject.DestroyImmediate(iconFont);
+
+                            iconFont = null;
+                        }
+                    }
+
+                    if (iconFont == null)
+                    {
+                        Initialize();
+                        // If it's null at this point, generate it now
+                        iconFont = Font.CreateDynamicFontFromOSFont(OSFontName, FontSize);
+                        iconFont.name = OSFontName;
+                        // Pre-load the values set in the profile
+                        iconFont.RequestCharactersInTexture(charactersInFont);
+                    }
+                }
+
+                return iconFont;*/
+            }
+            set
+            {
+                if (iconFont != value)
+                {
+                    // We'll need to re-initialize
+                    initialized = false;
+                    iconFont = value;
+                }
+            }
+        }
+
+        public override List<string> GetIconKeys()
+        {
+            Initialize();
+
+            return new List<string> (UnicodeKeys);
+        }
+
+        public override bool GetIcon(string iconName, MeshRenderer targetRenderer, MeshFilter targetMesh, bool useDefaultIfNotFound)
+        {
+            Initialize();
+
+            if (targetRenderer == null || targetMesh == null)
+            {
+                return false;
+            }
+
+            Texture2D icon = null;
+            if (useDefaultIfNotFound)
+            {
+                icon = _IconNotFound_;
+            }
+
+            bool useDefaultMesh = true;
+
+            if (IconFont != null)
+            {
+                if (!string.IsNullOrEmpty(iconName))
+                {
+                    if (GetIntValueFromHex (iconName) == 0)
+                    {
+                        if (useDefaultIfNotFound)
+                        {
+                            icon = _IconNotFound_;
+                        }
+                        return icon != null;
+                    }
+                    string textValue = GetCharStringFromHex(iconName);
+                    // If we've gotten this far, there's no need to use a default mesh
+                    useDefaultMesh = true;
```

Comment:
redundant setting of `useDefaultMesh`.  The value is never updated when we set it on 142.